pyk/src/tests/integration/proof/test_mini_kevm.py

```diff
@@ -106,3 +168,87 @@ def test_all_path_reachability_prove(
 
             assert proof.status == proof_status
             assert leaf_number(proof) == expected_leaf_number
+
+    @pytest.mark.parametrize(
+        'test_id,spec_file,spec_module,claim_id,max_iterations,max_depth,cut_rules,proof_status,expected_leaf_number',
+        CUSTOM_STEP_TEST_DATA,
+        ids=[test_id for test_id, *_ in CUSTOM_STEP_TEST_DATA],
+    )
+    def test_custom_step(
```

Comment:
I think you could use something much simpler than `MINI-KEVM` for testing this. Something like

```k
module CUSTOM-STEP
    sort State ::= "a" | "b" | "c" [token]
    rule <k> a => b </k>
endmodule
```

Then add a custom step to `c`.